--- Makefile_config	2020-05-17 21:22:03.757259571 +0200
+++ Makefile_config_mod	2020-05-17 21:24:47.777265294 +0200
@@ -35,7 +35,7 @@
 	HOSTCFLAGS += -I/usr/local/include/ncurses -DCURSES_LOC="<ncurses/curses.h>"
 else
 ifeq (/usr/include/ncurses.h, $(wildcard /usr/include/ncurses.h))
-	HOSTNCURSES += -DCURSES_LOC="<ncurses.h>"
+	HOSTNCURSES += -DCURSES_LOC="<ncurses.h>" -fcommon
 else
 	HOSTNCURSES += -DCURSES_LOC="<curses.h>"
 endif
