--- std-gnu11.m4	2021-01-14 15:46:15.482423011 +0100
+++ std-gnu11.m4_mod	2021-01-14 15:57:29.247075434 +0100
@@ -27,6 +27,8 @@
 # François Pinard, Karl Berry, Richard Pixley, Ian Lance Taylor,
 # Roland McGrath, Noah Friedman, david d zuhn, and many others.
 
+m4_version_prereq([2.70], [], [
+
 
 # AC_PROG_CC([COMPILER ...])
 # --------------------------
@@ -822,3 +824,5 @@
 dnl with extended modes being tried first.
 [[-std=gnu++11 -std=c++11 -std=gnu++0x -std=c++0x -qlanglvl=extended0x -AA]], [$1], [$2])[]dnl
 ])# _AC_PROG_CXX_CXX11
+
+])# m4_version_prereq
