--- Makefile_utils.orig	2022-03-18 15:52:04.000000000 +0000
+++ Makefile_utils	2022-03-18 12:43:17.000000000 +0000
@@ -54,6 +54,7 @@
 ifneq ($(RTCONFIG_BCMWL6),y)
 LDFLAGS += -L$(TOP)/nvram -L$(INSTALLDIR)/nvram/usr/lib -lnvram
 LDFLAGS += -L$(TOP)/shared -L$(INSTALLDIR)/shared/usr/lib -lshared
+LDFLAGS += -L$(TOOLCHAIN)/mipsel-brcm-linux-uclibc/sysroot/lib -lgcc_s
 endif
 
 #UTILS := mii-tool robocfg ttcp
