--- Makefile_networkmap.orig	2022-03-18 15:52:04.000000000 +0000
+++ Makefile_networkmap	2022-03-18 12:43:17.000000000 +0000
@@ -5,7 +5,7 @@
 
 CFLAGS  += -Wall -DASUS -DBCMNVRAM -I. -I$(TOP)/shared -I$(TOP)/kernel_header/include -I$(SRCBASE)/include
 CFLAGS  += -s -O2 -DNO_PARALLEL
-LDFLAGS += -L$(TOP)/shared -lshared -L$(TOP)/nvram${BCMEX} -lnvram 
+LDFLAGS += -L$(TOP)/shared -lshared -L$(TOP)/nvram${BCMEX} -lnvram  -L$(TOOLCHAIN)/$(SUBARCH)-brcm-linux-uclibc/sysroot/lib -lgcc_s 
 ifeq ($(RTCONFIG_BCMARM),y)
 LDFLAGS += -lgcc_s
 endif
