--- Makefile_arch_mips_orig	2022-01-22 15:33:38.000000000 +0000
+++ Makefile_arch_mips	2022-01-15 14:50:35.000000000 +0000
@@ -89,6 +89,15 @@
 # names.
 cflags-y += -ggdb
 
+#
+# pass -msoft-float to GAS if it supports it.  However on newer binutils
+# (specifically newer than 2.24.51.20140728) we then also need to explicitly
+# set ".set hardfloat" in all files which manipulate floating point registers.
+#
+ifneq ($(call as-option,-Wa$(comma)-msoft-float,),)
+	cflags-y		+= -DGAS_HAS_SET_HARDFLOAT -Wa,-msoft-float
+endif
+
 cflags-y += -ffreestanding
 
 #
