--- mem_utils.c_orig	2022-01-27 15:12:09.000000000 +0000
+++ mem_utils.c	2021-12-31 11:30:09.000000000 +0000
@@ -16,7 +16,7 @@
  *
  * Returns: Pointer to freshly allocated memory
  */
-inline void *xmalloc(size_t size) {
+extern inline void *xmalloc(size_t size) {
 	void *ptr;
 	ptr = malloc(size);
 	if (ptr == NULL) {
@@ -34,7 +34,7 @@
  *
  * Returns: Pointer to reallocated memory
  */
-inline void *xrealloc(void *inptr, size_t size) {
+extern inline void *xrealloc(void *inptr, size_t size) {
 	void *ptr;
 	ptr = realloc(inptr, size);
 	if (ptr == NULL) {
