--- platform.mak.orig	2022-03-18 15:52:04.000000000 +0000
+++ platform.mak	2022-03-18 12:43:17.000000000 +0000
@@ -14,6 +14,5 @@
 export BCMEX := _arm
 export EXTRA_FLAG := -lgcc_s
 export ARCH := arm
 export HOST := arm-linux
 export TOOLS := $(SRCBASE)/toolchains/hndtools-arm-linux-2.6.36-uclibc-4.5.3
 export RTVER := 0.9.32.1
@@ -25,15 +25,15 @@
 
 export KERNEL_BINARY=$(LINUXDIR)/arch/mips/brcm-boards/bcm947xx/compressed/zImage
 export PLATFORM := mipsel-uclibc
-export CROSS_COMPILE := mipsel-uclibc-
+export CROSS_COMPILE := mipsel-brcm-linux-uclibc-
 export CROSS_COMPILER := $(CROSS_COMPILE)
-export READELF := mipsel-linux-readelf
+export READELF := mipsel-brcm--linux-readelf
 export CONFIGURE := ./configure --host=mipsel-linux --build=$(BUILD)
 export HOSTCONFIG := linux-mips32
 export ARCH := mips
 export HOST := mipsel-linux
-export TOOLS := $(SRCBASE)/../../tools/brcm/hndtools-mipsel-linux
-export RTVER := 0.9.30.1
+export TOOLS := $(SRCBASE)/../../tools/brcm/K26/hndtools-mipsel-uclibc-5.3/usr/
+export RTVER := 0.9.32.1
 endif
 EXTRA_CFLAGS := -DLINUX26 -DCONFIG_BCMWL5 -DDEBUG_NOISY -DDEBUG_RCTEST -pipe -DTTEST
 
