--- etc47xx.c.orig	2021-12-30 19:54:43.433929000 +0100
+++ etc47xx.c	2021-12-27 15:52:55.185567800 +0100
@@ -165,7 +165,7 @@
 	bcmenetregs_t *regs;
 	char name[16];
 	char *var;
-	uint boardflags, boardtype;
+	uint boardflags;
 
 	ET_TRACE(("et%d: chipattach: regsva 0x%lx\n", etc->unit, (ulong)regsva));
 
@@ -214,7 +214,6 @@
 	etc->boardflags = getintvar(ch->vars, "boardflags");
 
 	boardflags = etc->boardflags;
-	boardtype = ch->sih->boardtype;
 
 	/* get our local ether addr */
 	sprintf(name, "et%dmacaddr", etc->coreunit);
