--- etc_adm.c.orig	2021-12-30 19:54:43.433929000 +0100
+++ etc_adm.c	2021-12-27 17:21:20.465567800 +0100
@@ -108,14 +108,14 @@
 static void
 adm_enable(adm_info_t *adm)
 {
-	void *regs;
+	/*void *regs; */
 
 	/* Save current core index */
 	adm->coreidx = si_coreidx(adm->sih);
 
 	/* Switch to GPIO core for faster access */
-	regs = si_gpiosetcore(adm->sih);
-	ASSERT(regs);
+	/*regsx = si_gpiosetcore(adm->sih); */
+	ASSERT(si_gpiosetcore(adm->sih));
 }
 
 /* Disable register access to the chip */
