--- Makefile_cstats_arm.orig	2022-03-14 15:09:48.000000000 +0000
+++ Makefile_cstats_arm	2022-03-12 17:22:27.000000000 +0000
@@ -3,7 +3,7 @@
 CFLAGS = -Wall -Os $(EXTRACFLAGS)
 CFLAGS += -I. -I$(TOP)/shared -I$(SRCBASE)/include -idirafter$(LINUXDIR)/include
 #CFLAGS += -ffunction-sections -fdata-sections
-LDFLAGS	= -L../nvram$(BCMEX) -lnvram -L../shared -lshared $(LIBS)
+LDFLAGS = -L$(TOP)/nvram$(BCMEX) -lnvram -L$(TOP)/shared -lshared -L$(TOOLCHAIN)/arm-brcm-linux-uclibcgnueabi/sysroot/lib -lgcc_s  -L$(TOOLCHAIN)/lib/gcc/arm-brcm-linux-uclibcgnueabi/5.3.0
 
 ifeq ($(RTCONFIG_BCMARM),y)
 LIBS = -lgcc_s
